extensions/binding/org.eclipse.smarthome.binding.hue/src/main/java/org/eclipse/smarthome/binding/hue/handler/HueBridgeHandler.java

```diff
@@ -69,75 +73,25 @@
  * @author Stefan Bußweiler - Added new thing status handling
  * @author Jochen Hiller - fixed status updates, use reachable=true/false for state compare
  * @author Denis Dudnik - switched to internally integrated source of Jue library
+ * @author Samuel Leisering - added Sensor support
  */
 @NonNullByDefault
 public class HueBridgeHandler extends ConfigStatusBridgeHandler implements HueClient {
 
-    private static final String LIGHT_STATE_ADDED = "added";
-
-    private static final String LIGHT_STATE_CHANGED = "changed";
-
-    public static final Set<ThingTypeUID> SUPPORTED_THING_TYPES = Collections.singleton(THING_TYPE_BRIDGE);
-
-    private static final int DEFAULT_POLLING_INTERVAL = 10; // in seconds
-
-    private static final String DEVICE_TYPE = "EclipseSmartHome";
-
-    private final Logger logger = LoggerFactory.getLogger(HueBridgeHandler.class);
-
-    private final Map<String, FullLight> lastLightStates = new ConcurrentHashMap<>();
-
-    private boolean lastBridgeConnectionState = false;
-
-    private boolean propertiesInitializedSuccessfully = false;
-
-    private final List<LightStatusListener> lightStatusListeners = new CopyOnWriteArrayList<>();
-
-    private @Nullable ScheduledFuture<?> pollingJob;
-
-    @NonNullByDefault({})
-    private HueBridge hueBridge = null;
-
-    private final Runnable pollingRunnable = new Runnable() {
+    ReentrantLock pollingLock = new ReentrantLock();
 
+    abstract class PollingRunnable implements Runnable {
```

Comment:
While touching thus anyways, do you think it is possible to get rud of the `Runnable` interface and find a more elegant solution?

Reply:
the polling runnables are scheduled using the `BaseThingHandlers` executor service, so i'm not sure how to improve this. Do you have something in mind?

Reply:
I think it's connected to this comment: https://github.com/eclipse/smarthome/pull/5996#discussion_r207131819

Maybe you should try to register at the hue bridge and get notified if the value on the hue bridge changed. I don't know if this is possible, but it would be an elegant way to solve the runnable-problem.

Reply:
I see. Unfortunately the bridge does not support registering for changes, so polling is the only way currently

Reply:
> Do you have something in mind?

Yes, I do. Refactoring anonymous classes to Java 8 lambda expressions. This will make the code easier to maintain and more readable.

Currently we are using something like this:
```java
// using an anonymous class
Runnable pollingRunnable = new Runnable() {
    @Override
    public void run() {
        // do polling stuff
    }
};

scheduler.scheduleWithFixedDelay(pollingRunnable, 1, 1, TimeUnit.SECONDS);
```
Possible replacement:
```java
// using a lambda expression
scheduler.scheduleWithFixedDelay(() -> {
    // do polling stuff
}, 1, 1, TimeUnit.SECONDS);
```
Or:
```java
class myClass {
    public void poll() {
        // do polling stuff
    }

    public void myFunction() {
        scheduler.scheduleWithFixedDelay(this::poll, 1, 1, TimeUnit.SECONDS);
    }
}
```